.github/workflows/java-cd.yml

```diff
@@ -158,3 +143,109 @@ jobs:
+
+            - name: Upload artifacts to publish
+              uses: actions/upload-artifact@v4
+              with:
+                name: valkey-${{ env.RELEASE_VERSION }}
+                path: |
+                  build
+
+            - name: Publish to Maven Central
+              id: maven-deployment
+              run: |
+                  export DEPLOYMENT_ID=`curl --request POST \
+                  -u "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" \
+                  --form bundle=@build.zip \
+                  https://central.sonatype.com/api/v1/publisher/upload | tail -n 1`
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_ENV
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_OUTPUT
+                  echo $DEPLOYMENT_ID
+
+            - name: Check status of deployment
+              run: |
+                  sleep 20
+                  export DEPLOYMENT_STATUS=`curl --request POST \
+                  -u "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" \
+                  "https://central.sonatype.com/api/v1/publisher/status?id=${{ env.DEPLOYMENT_ID }}" \
+                  | jq '.deploymentState'`
+                  echo $DEPLOYMENT_STATUS
+                  if [[ $DEPLOYMENT_STATUS != ""\"VALIDATED"\"" || $DEPLOYMENT_STATUS -eq null ]]; then exit 1 ; fi
+
+            - uses: actions/checkout@v4
+              with:
+                submodules: recursive
+
+            - name: Set up JDK
+              uses: actions/setup-java@v4
+              with:
+                distribution: "temurin"
+                java-version: '11'
+
+            - name: Install protoc (protobuf)
+              uses: arduino/setup-protoc@v3
+              with:
+                version: "26.1"
+                repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Test deployment
+              working-directory: java
+              run: |
+                  export ORG_GRADLE_PROJECT_centralManualTestingAuthHeaderName="Authorization"
+                  export ORG_GRADLE_PROJECT_centralManualTestingAuthHeaderValue="${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}"
+                  export RUN_DEPLOYMENT_TEST=true
+                  export GLIDE_RELEASE_VERSION=${{ env.RELEASE_VERSION }}
+                  ./gradlew run
+
+            - name: Drop deployment if validation fails
+              if: ${{ failure() }}
+              run: |
+                  curl --request DELETE \
+                  -u "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" \
+                  "https://central.sonatype.com/api/v1/publisher/deployment/${{ env.DEPLOYMENT_ID }}"
+
+    publish-release-to-maven:
+        if: ${{ inputs.maven_publish == true || github.event_name == 'push' }}
+        needs: [publish-to-maven-central-deployment]
+        runs-on: ubuntu-latest
+        environment: AWS_ACTIONS
```

Comment:
Can you test it on another workflow to confirm that it will wait for the second approval?

Reply:
It does work. I ran a workflow where it got to this point and required approval, however we did not approve as we did not want to push to Maven just yet.